co/dataIStream.h

```diff
@@ -217,12 +217,17 @@ class DataIStream
     template <class T>
     void _read(T& value, const boost::false_type&)
     {
-        _readSerializable(value, boost::is_base_of<servus::Serializable, T>());
+        boost::is_base_of<servus::Serializable, T> base_of = boost::is_base_of<servus::Serializable, T>();
+        _readSerializable(value, &base_of);
     }
 
     /** Read a serializable object. */
     template <class T>
-    void _readSerializable(T& value, const boost::true_type&);
+    void _readSerializable(T& value, const boost::true_type*);
+
+    /** Catch for non-serializable object. */
+    template <class T>
+    void _readSerializable(T&, const boost::false_type*);
```

Comment:
Why is this needed? Compilation should fail if a non-Serializable is read, no?

Reply:
Compilation was failing - this is why I added it. Perhaps that is indicative of an issue elsewhere in the code if this should never happen?

Reply:
Yes, that's what I'm thinking.

Reply:
Ok - is this something you would need me to try and track down, or will you be taking a look?

Reply:
Hi - I'm just following up to see if there is a fix needed on my end, or what the next steps are? Equalizer (which uses Collage) seems like a great option for the work I'm trying to do, but it just won't compile (or perhaps has a different issue with the suggested patch), which means I may unfortunately need to abandon pursuing its use in favor of a different distributed rendering framework